doc/adrs/107_add_support_for_transient_data_in_forms.adoc

```diff
@@ -0,0 +1,84 @@
+= [ADR-107] Link widgets together in forms
+
+== Context
+
+The forms that can be defined in Sirius Web are sharing a core restriction with the form from EEF.
+In both case, you can only create forms using data backed up by models.
+As a result, there are families of use cases that are complex or borderline impossible to support.
+In some use cases, the end users may want to manipulate, in their forms, transient data which are not persisted in a model.
+
+Among those use case, we can find a dialog to create a new element.
+We don't want to create the element directly when we open the dialog and edit its properties.
+We want to create the element only when the user will have click on "Ok".
+
+Another popular use case is the creation of master / details interfaces.
+In this case, we could have a select showing a list of objects and below various widgets displaying the properties of the selected object.
+While the data in the various widgets would come from the model, the currently selected object should not be persisted in the model at all.
+
+
+== Decision
+
+We will add the ability for specifiers to create forms allowing the manipulation of transient data not backup by a model.
+
+Let's consider a very simple text field allowing the manipulation of a transient piece of text.
+
+In order to allow such use case, we will have to specify how to retrieve the value of the textfield.
+For that, we will keep everything we have with the variable manager and the value provider (for the core API) and the value expression (for the view DSL).
+We will only have to provide some additional variables to retrieve the transient data and we will be able to display some text in our text field.
+
+We will thus have to figure out how to define, initialize and update those variables.
+Since we want those variables to be transient, they will be specific to every single instance of a representation.
+For example, two users watching the same form with a master / details user interface could have selected a different value.
+
+=== Transient datastore and lifecycle of the representation
+
+We could store those transient data in the frontend.
+This woule easily ensure that they are specific to the instance of the representation.
+They would be provided to the backend as subscription parameters using the input in a similar fashion as the list of expanded tree items in a tree-based representation.
+
+Such an approach would have a couple of downsides.
+Firstly, we would be limited to basic types of transient data (string, boolean, number, list) and it would thus not be possible to have transient model objects or custom types.
```

Comment:
Not sure this is a problem.
The values of the variables would correspond to the "visual state" of the widgets (e.g. for a text field, the text it currently displays; for a list the (list of) ids of the currently selected elements etc.).
It would be up to the backend to interpret those in more structured ways.

Reply:
I'm not sure that it's a major issue too but it would be more complex to handle some use cases (for example, if the number of widgets to display is dynamic, we would thus need a dynamic number of transient variables).

If I want to build a new class dialog to create an UML Class (name, is abstract, implemented interfaces, extended class) with some attributes (name, cardinality, type) all at once that would be quite complex. If I could have a transient UML class it would be way easier to store / retrieve everything (especially the properties of all the attributes).